### PR TITLE
Implement upgrade to capella

### PR DESCRIPTION
**Pull request by potuz**
This PR implements `upgrade_to_capella` as defined in the specification for the Capella Fork.

### CODE REVIEW

File: beacon-chain/core/time/slot_epoch.go [hunk 1]
```diff
@@ -70,6 +70,15 @@ func CanUpgradeToBellatrix(slot types.Slot) bool {
 	return epochStart && bellatrixEpoch
 }
 
+// CanUpgradeToCapella returns true if the input `slot` can upgrade to Capella.
+// Spec code:
+// If state.slot % SLOTS_PER_EPOCH == 0 and compute_epoch_at_slot(state.slot) == CAPELLA_FORK_EPOCH
+func CanUpgradeToCapella(slot types.Slot) bool {
```
Comment (rkapka) [inline beacon-chain/core/time/slot_epoch.go:76]: Missing test:
```
func TestCanUpgradeToCapella(t *testing.T) {
	params.SetupTestConfigCleanup(t)
	bc := params.BeaconConfig()
	bc.CapellaForkEpoch = 5
	params.OverrideBeaconConfig(bc)
	tests := []struct {
		name string
		slot types.Slot
		want bool
	}{
		{
			name: "not epoch start",
			slot: 1,
			want: false,
		},
		{
			name: "not capella epoch",
			slot: params.BeaconConfig().SlotsPerEpoch,
			want: false,
		},
		{
			name: "capella epoch",
			slot: types.Slot(params.BeaconConfig().CapellaForkEpoch) * params.BeaconConfig().SlotsPerEpoch,
			want: true,
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := time.CanUpgradeToCapella(tt.slot); got != tt.want {
				t.Errorf("CanUpgradeToCapella() = %v, want %v", got, tt.want)
			}
		})
	}
}
```

File: beacon-chain/state/state-native/getters_payload_header.go [hunk 2]
```diff
@@ -14,10 +14,6 @@ func (b *BeaconState) LatestExecutionPayloadHeader() (interfaces.ExecutionData,
 		return nil, errNotSupported("LatestExecutionPayloadHeader", b.version)
 	}
 
-	if b.latestExecutionPayloadHeader == nil {
-		return nil, nil
-	}
-
```
Comment (rkapka) [inline beacon-chain/state/state-native/getters_payload_header.go:20]: Why is this removed?

File: consensus-types/blocks/execution.go [hunk 3]
```diff
@@ -140,11 +140,21 @@ func (e executionPayload) Transactions() ([][]byte, error) {
 	return e.p.Transactions, nil
 }
 
+// TransactionsRoot --
```
Comment (rkapka) [inline consensus-types/blocks/execution.go:143]: There are no tests for this file. I have some in [`develop...capella-state-processing`#diff-a4ffcfb9e9](https://github.com/prysmaticlabs/prysm/compare/develop...capella-state-processing#diff-a4ffcfb9e9d5ea43d14f3217b54c9d2502a8b33d05934d07628b07ff79394887)

File: beacon-chain/state/state-native/getters_payload_header.go (hunk 2 above)
Comment (potuz) [reply beacon-chain/state/state-native/getters_payload_header.go:20]: This one makes a lot of runtime changes on Capella, it took me a while to track down, the issue is that for Capella states this is always nil! The option is to case test for the fork version, but I think removing here is better.

**Review by rkapka — APPROVED** [review]
